File: src/main/java/org/myfintech/payment/domain/ClientCreateDTO.java
/*
 * Copyright (c) 2024 MyFintech Payment Service
 * All rights reserved.
 *
 * This software is proprietary and confidential. Unauthorized copying of this file,
 * via any medium, is strictly prohibited.
 * @author : Dhanuka Ranasinghe
 * @since : Date: 11/07/2025
 */
package org.myfintech.payment.domain;

import jakarta.validation.constraints.NotBlank;
import static org.myfintech.payment.validator.PaymentValidationConstants.*;

public record ClientCreateDTO(@NotBlank(message = CLIENT_NAME_REQUIRED) String clientName) {}

File: src/main/kotlin/org/myfintech/payment/domain/ClientCreateDTO.kt
/*
 * Copyright (c) 2024 MyFintech Payment Service
 * All rights reserved.
 *
 * This software is proprietary and confidential. Unauthorized copying of this file,
 * via any medium, is strictly prohibited.
 * @author : Dhanuka Ranasinghe
 * @since : Date: 11/07/2025
 */
package org.myfintech.payment.domain

import jakarta.validation.constraints.NotBlank
import org.myfintech.payment.validator.PaymentValidationConstants

@JvmRecord
data class ClientCreateDTO(val clientName: @NotBlank(message = PaymentValidationConstants.CLIENT_NAME_REQUIRED) String)
